packages/xo-web/src/xo-app/new/sr/index.js

```diff
@@ -222,22 +221,46 @@ export default class New extends Component {
       path: undefined,
       paths: undefined,
       type: undefined,
-      unused: undefined,
       usage: undefined,
-      used: undefined,
       zfsPools: undefined,
     }
-    this.getHostSrs = createFilter(
-      () => this.props.srs,
-      createSelector(
-        () => this.state.host,
-        ({ $pool, id }) => sr => sr.$container === $pool || sr.$container === id
-      ),
-      true
-    )
   }
 
-  _handleSubmit = async () => {
+  getHostSrs = createFilter(
+    () => this.props.srs,
+    createSelector(
+      () => this.state.host,
+      host => host !== undefined && (sr => sr.$container === host.$pool || sr.$container === host.id)
+    )
+  )
+
+  getUsedSrs = createCollectionWrapper(
+    createSelector(
+      this.getHostSrs,
+      () => this.state.existingSrs,
+      (hostSrs, existingSrs) => {
+        if (existingSrs === undefined) {
+          return
+        }
+
+        const usedSrs = []
+        existingSrs.forEach(({ uuid }) => {
```

Comment:
To not reach `existingSr` two times,`getHostSrs` can return `used` and `unused` SRs.

Reply:
True, but making a selector return a collection of collections would make the selector's cache useless.